serde_derive/src/de.rs

```diff
@@ -1669,12 +1706,12 @@ fn deserialize_untagged_enum(
                 params,
                 variant,
                 cattrs,
-                quote!(
-                    _serde::__private::de::ContentRefDeserializer::<__D::Error>::new(&__content)
-                ),
+                deserializer.clone(),
             ))
         });
-
+    let attempts = first_attempt(deserializer.clone())
```

Comment:
```suggestion
    let attempts = first_attempt(deserializer)
```

the last use can consume it.

Reply:
I tried this, but `deserializer` is actually captured by the closure in `map` producing an iterator over the rest of the attempts.